third_party/move/move-bytecode-verifier/src/type_safety.rs

```diff
@@ -300,6 +301,115 @@ fn call(
     Ok(())
 }
 
+fn clos_eval(
+    verifier: &mut TypeSafetyChecker,
+    meter: &mut impl Meter,
+    offset: CodeOffset,
+    expected_ty: &SignatureToken,
+) -> PartialVMResult<()> {
+    let SignatureToken::Function(param_tys, ret_tys, abilities) = expected_ty else {
+        // The signature checker has ensured this is a function
+        safe_assert!(false);
```

Comment:
Why
```
safe_assert!(false);
unreachable!()
```
can't we just return an error? The pattern seems very weird...

Reply:
Didn't see the definition of `safe_assert` is to panic with debug assertions, I thought it is always an error. I think the name could have been better, e.g., `debug_panic_or_error()`, but whatever